tests-manual/update_test_data.py

```diff
@@ -0,0 +1,300 @@
+import argparse
+import glob
+import json
+from collections import defaultdict
+from json.decoder import JSONDecodeError
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+from deepdiff import DeepDiff
+
+from ingest_validation_tools.cli_utils import dir_path
+from ingest_validation_tools.error_report import ErrorReport
+from ingest_validation_tools.table_validator import ReportType
+from ingest_validation_tools.upload import Upload
+from ingest_validation_tools.validation_utils import TSVError, read_rows
+from tests.test_dataset_examples import (
+    DATASET_EXAMPLES_OPTS,
+    DATASET_IEC_EXAMPLES_OPTS,
+    MockException,
+    TestDatasetExamples,
+    clean_report,
+    dataset_test,
+    diff_test,
+)
+
+
+class UpdateData:
+
+    def __init__(
+        self,
+        dir: str,
+        globus_token: str,
+        exclude: List = [],
+        opts: Dict = {},
+        verbose: bool = False,
+        dry_run: bool = True,
+    ):
+        self.dir = dir
+        self.globus_token = globus_token
+        self.exclude = exclude
+        self.opts = opts if opts else DATASET_EXAMPLES_OPTS
+        self.verbose = verbose
+        self.upload_verbose = True if "plugin-tests" in dir else False
+        self.dry_run = dry_run
+
+    def log(self, verbose_message, short_message: Optional[str] = None):
+        if self.verbose:
+            print(verbose_message)
+        elif short_message:
+            print(short_message)
+
+    def update_test_data(self) -> Dict[str, List]:
+        print(f"Evaluating {self.dir}...")
+        self.change_report = defaultdict(list)
+        self.upload = Upload(
+            Path(f"{self.dir}/upload"),
+            globus_token=self.globus_token,
+            **self.opts,
+            verbose=self.upload_verbose,
+        )
+        info = self.upload.get_info()
+        errors = self.upload.get_errors()
+        report = ErrorReport(info=info, errors=errors)
+        if "Too many requests" in report.as_md():
+            raise Exception(
+                f"Something went wrong with Spreadsheet Validator request for {self.dir}."
+            )
+        if "fixtures" not in self.exclude:
+            new_data = self.update_fixtures(report)
+            fixtures = self.open_or_create_fixtures()
+            diff = DeepDiff(fixtures, new_data, ignore_order=True, report_repetition=True)
```

Comment:
@jpuerto-psc So it turns out the `==` was doing a pretty thorough job with nested dicts, but it would absolutely get tripped up if there was any difference in list ordering. I think list ordering is not meaningful here so I have set it to `False` (while making sure to catch any duplicates in the list, which are ignored by default if you ignore order). This also lets us get a better diff output, which I think is nice. What do you think?

Reply:
That makes sense to me! I agree that list ordering is not relevant here and that set equivalence is more important.